src/mapping/nav2_map_server/test/unit/test_occ_grid.cpp

```diff
@@ -0,0 +1,143 @@
+#include "test_constants/test_constants.h"
+
+#ifdef TEST_DIRECTORY
+#define TEST_DIR TEST_DIRECTORY
+#else
+#define TEST_DIR "PleaseSetDirectory"
+#endif
+
+using namespace std; // NOLINT
+using std::experimental::filesystem::path;
+
+class OccGridTest : public OccGridLoader
+{
+public:
+  OccGridTest() {}
+  ~OccGridTest() {}
+
+  nav_msgs::srv::GetMap::Response getMap() {return occ_resp_;}
+};
+
+class MapServerTest : public ::testing::Test
+{
+protected:
+  OccGridTest * OccTest;
+
+  MapServerTest()
+  {
+    OccTest = new OccGridTest;
+  }
+};
+
+/* Try to load a valid PNG file.  Succeeds if no exception is thrown, and if
+ * the loaded image matches the known dimensions and content of the file.
+ *
+ * This test can fail on OS X, due to an apparent limitation of the
+ * underlying SDL_Image library. */
+
+TEST_F(MapServerTest, loadValidPNG)
+{
+  try {
+    nav_msgs::srv::GetMap::Response map_resp;
+    auto test_yaml = path(TEST_DIR) / path(g_valid_yaml_file);
+    auto test_png = path(TEST_DIR) / path(g_valid_png_file);
+    OccTest->loadMapInfoFromFile(test_yaml.string());
+    OccTest->loadMapFromFile(test_png.string());
+    OccTest->setMap();
+    map_resp = OccTest->getMap();
+
+    EXPECT_FLOAT_EQ(map_resp.map.info.resolution, g_valid_image_res);
+    EXPECT_EQ(map_resp.map.info.width, g_valid_image_width);
+    EXPECT_EQ(map_resp.map.info.height, g_valid_image_height);
+    for (unsigned int i = 0; i < map_resp.map.info.width * map_resp.map.info.height; i++) {
+      EXPECT_EQ(g_valid_image_content[i], map_resp.map.data[i]);
+    }
+  } catch (...) {
+    ADD_FAILURE() << "Uncaught exception : " <<
+      "This is OK on OS X";
+  }
+}
+
+/* Try to load a valid BMP file.  Succeeds if no exception is thrown, and if
+ * the loaded image matches the known dimensions and content of the file. */
+
+TEST_F(MapServerTest, loadValidBMP)
+{
+  try {
+    nav_msgs::srv::GetMap::Response map_resp;
+    auto test_yaml = path(TEST_DIR) / path(g_valid_yaml_file);
+    auto test_png = path(TEST_DIR) / path(g_valid_png_file);
+    OccTest->loadMapInfoFromFile(test_yaml.string());
+    OccTest->loadMapFromFile(test_png.string());
+    OccTest->setMap();
+    map_resp = OccTest->getMap();
+
+    EXPECT_FLOAT_EQ(map_resp.map.info.resolution, g_valid_image_res);
+    EXPECT_EQ(map_resp.map.info.width, g_valid_image_width);
+    EXPECT_EQ(map_resp.map.info.height, g_valid_image_height);
+    for (unsigned int i = 0; i < map_resp.map.info.width * map_resp.map.info.height; i++) {
+      EXPECT_EQ(g_valid_image_content[i], map_resp.map.data[i]);
+    }
+  } catch (...) {
+    ADD_FAILURE() << "Uncaught exception";
+  }
+}
+
+TEST_F(MapServerTest, loadInvalidFile)
+{
+  try {
+    auto test_yaml = path(TEST_DIR) / path(g_valid_yaml_file);
+    auto test_png = path(TEST_DIR) / path("foo");
+    OccTest->loadMapInfoFromFile(test_yaml.string());
+    OccTest->loadMapFromFile(test_png.string());
+  } catch (std::runtime_error & e) {
```

Comment:
There is a googletest macro to check for an exception being thrown ... something like ASSERT_THROW